### PR TITLE
Add heartbeatFrequencyMS default value for retryable read/writes and transaction tests.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**
Questions:
1. Should the same change be added for transactions-convenient-api?
2. Should I create a DRIVER ticket?

### CODE REVIEW

File: source/retryable-reads/tests/README.rst
Comment (jmikola) [inline source/retryable-reads/tests/README.rst:160]: I'm on the fence about sprinkling this throughout all of the test files. It seems like something that would be better documented in a "best practices" document at the top of the repository, since it might as well apply to any test runner working with a server. Drivers would also probably want to do this in their own test suites (I've done so in my PHPUnit configuration for a few years now).

Comment (DmitryLukyanov) [reply source/retryable-reads/tests/README.rst:160]: >a "best practices" document
do you mean this doc: https://github.com/mongodb/specifications/blob/master/README.rst ?

Comment (jmikola) [reply source/retryable-reads/tests/README.rst:160]: The main README may be the best place for now. I'll defer to @jyemin and @ShaneHarvey, though.

My main concern is wanting to avoid duplicating the same text in multiple places.

Comment (ShaneHarvey) [reply source/retryable-reads/tests/README.rst:160]: I think this guidance would be out of place in the main readme. What if we add the description of "Speeding Up Tests" to only the transaction spec file and then link to that section in the other files?

Comment (DmitryLukyanov) [reply source/retryable-reads/tests/README.rst:160]: sgtm, @jmikola ? :)

Comment (jmikola) [reply source/retryable-reads/tests/README.rst:160]: I don't object. I'd also suggest noting this in SPEC ticket and linking it up to [DRIVERS-709](https://jira.mongodb.org/browse/DRIVERS-709). From there, I can include it in the final README for the unified test runner.

**Comment by DmitryLukyanov** [comment]
Ready for review :)

File: source/retryable-reads/tests/README.rst
Comment (DmitryLukyanov) [reply source/retryable-reads/tests/README.rst]: done

File: source/retryable-reads/tests/README.rst
Comment (jmikola) [inline source/retryable-reads/tests/README.rst:151]: This should line up with the heading text, but I also don't see any reason this needs to change. The README already had a section for this, so we can leave it at its current nesting level, since this applies within the "Network Error Tests" section.

File: source/transactions/tests/README.rst
Comment (jmikola) [inline source/transactions/tests/README.rst:83]: Reading this alongside the existing paragraph on tweaking `minHeartbeatFrequencyMS` in Retryable Reads, how would drivers be able to tweak `heartbeatFrequencyMS` if `minHeartbeatFrequencyMS` is already [fixed at 500ms](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-monitoring.rst#minheartbeatfrequencyms)?

I think this text should state that drivers manually lower `minHeartbeatFrequencyMS` if possible _and_ reduce `heartbeatFrequencyMS` to that same value -- with an exception if the test already has an explicit `heartbeatFrequencyMS` value. You can suggest 5ms for both if you wish.

That said, I'm not sure why we should bother adding anything to Transactions if we already have a section in the Retryable Reads spec. Perhaps just have everything in this PR refer to the Retryable Read section instead (and modify it as you like).

File: source/transactions/tests/README.rst
Comment (jmikola) [inline source/transactions/tests/README.rst:648]: I don't think any of these additions need a changelog entry, since we're just provide some advice and this not anything that requires driver changes.

File: source/retryable-reads/tests/README.rst
Comment (jmikola) [inline source/retryable-reads/tests/README.rst:161]: Since this document already talks about adjusting `minHeartbeatFrequencyMS`, I'd propose you modify it _and_ also suggest that drivers use a default `heartbeatFrequencyMS` value to match (which was never discussed here, but probably should have been).

File: source/transactions/tests/README.rst
Comment (ShaneHarvey) [reply source/transactions/tests/README.rst:648]: Agreed.

File: source/transactions/tests/README.rst
Comment (ShaneHarvey) [reply source/transactions/tests/README.rst:83]: > I think this text should state that drivers manually lower minHeartbeatFrequencyMS if possible and reduce heartbeatFrequencyMS to that same value -- with an exception if the test already has an explicit heartbeatFrequencyMS value. You can suggest 5ms for both if you wish.

Agreed. @DmitryLukyanov can you confirm that reducing both these values to 5ms does not cause any tests to fail?

Comment (DmitryLukyanov) [reply source/transactions/tests/README.rst:83]: @ShaneHarvey , I can confirm this: https://evergreen.mongodb.com/version/5f5ffde90ae6064926c283ff. 

`heartbeatFrequencyMS` has been reduced during the last month, for `minHeartbeatFrequencyMS`, I needed to make a code change. So, I'm not sure how strong required reducing minHeartbeatFrequencyMS, I would say not really

Comment (ShaneHarvey) [reply source/transactions/tests/README.rst:83]: Thanks for testing that. I second Jermey's request that we mention Drivers may reduce `minHeartbeatFrequencyMS`as well (if possible in the language). Note that we will not require drivers to reduce `minHeartbeatFrequencyMS`, nor force drivers to make `minHeartbeatFrequencyMS` configurable by users (because it shouldn't be). We can just suggest it as a possible way to speed up slow tests.

File: source/retryable-reads/tests/README.rst
Comment (jmikola) [reply source/retryable-reads/tests/README.rst:151]: @DmitryLukyanov ping

File: source/retryable-reads/tests/README.rst
Comment (jmikola) [inline source/retryable-reads/tests/README.rst:178]: This change description doesn't match up with the diff. I'd suggest:

> Suggest lowering heartbeatFrequencyMS in addition to minHeartbeatFrequencyMS

And please update the date.

File: source/retryable-writes/tests/README.rst
Comment (jmikola) [inline source/retryable-writes/tests/README.rst:97]: The retryable reads spec already has the most verbose explanation. Please link to that instead.

Also, the CWD of this file is `source/retryable-writes/tests/`, so the URL will likely need to be `../../retryable-reads/tests/README.rst#speeding-up-tests`.

File: source/retryable-writes/tests/README.rst
Comment (jmikola) [inline source/retryable-writes/tests/README.rst:346]: Per my [comment](https://github.com/mongodb/specifications/pull/820/files#r458261818) in the transactions README, this can be removed.

File: source/transactions/tests/README.rst
Comment (jmikola) [reply source/transactions/tests/README.rst:83]: Please make the changes in my earlier comment (https://github.com/mongodb/specifications/pull/820/files#r458262676) for retryable reads and change this to refer to that spec. I think the content can match what I suggested for retryable writes, which is just a reference to the retryable reads section.

Comment (DmitryLukyanov) [reply source/transactions/tests/README.rst]: done

**Comment by DmitryLukyanov** [comment]
NOTE: the new reviewers were added automatically. I didn't add them intentionally.

File: source/retryable-reads/tests/README.rst
Comment (jmikola) [inline source/retryable-reads/tests/README.rst:158]: The original paragraph still states 50ms for `minHeartbeatFrequencyMS`, which contradicts 5ms here.

I propose consolidating both paragraphs into one:

```
Drivers can greatly reduce the execution time of tests by setting `heartbeatFrequencyMS`_
and `minHeartbeatFrequencyMS`_ (internally) to a small value (e.g. 5ms), below what
is normally permitted in the SDAM spec. If a test specifies an explicit value for
heartbeatFrequencyMS (e.g. client or URI options), drivers SHOULD use that value.
```

File: source/retryable-reads/tests/README.rst
Comment (jmikola) [inline source/retryable-reads/tests/README.rst:161]: This can also use `../../server-discovery-and-monitoring/server-discovery-and-monitoring.rst#` as a base (before adding the anchors). No need to use absolute URLs.

File: source/retryable-reads/tests/README.rst
Comment (jmikola) [reply source/retryable-reads/tests/README.rst:158]: I don't think the anecdote about the Python driver using 50ms is particular relevant anymore, so I have no qualms about removing that. Also, I think "(internally)" is sufficient to explain that we're doing something that wouldn't be permitted by a driver's public API.

cc: @ShaneHarvey

Comment (DmitryLukyanov) [reply source/retryable-reads/tests/README.rst]: sgtm, done

File: source/retryable-reads/tests/README.rst
Comment (DmitryLukyanov) [reply source/retryable-reads/tests/README.rst]: done

**Review by jmikola — APPROVED** [review]
@isabelatkinson @durran I think your LGTMs may be required as code owners of transactions, but the changes are quite insignificant so it should be quick.

**Review by isabelatkinson — APPROVED** [review]

File: source/retryable-reads/tests/README.rst
Comment (ShaneHarvey) [inline source/retryable-reads/tests/README.rst:156]: drivers SHOULD use that value -> drivers MUST use that value

Comment (DmitryLukyanov) [reply source/retryable-reads/tests/README.rst]: done

**Review by ShaneHarvey — APPROVED** [review]

**Review by durran — APPROVED** [review]

**Pull request merged**
